File: Programmers/pro_12905_가장 큰 정사각형 찾기.py
def solution(board):
    W,H = len(board[0]),len(board)
    answer=max(board[0])
    for r in range(1,H):
        for c in range(1,W):
            if board[r][c]==1:
                board[r][c]=min(board[r-1][c],board[r][c-1],board[r-1][c-1])+1
                answer=max(answer,board[r][c])
    return answer**2

File: Programmers/pro_12905_가장큰정사각형찾기.js
function solution(board) {
    const [R, C] = [board.length, board[0].length]
    let answer = 0
    if (C === undefined) {
        for (let i = 0; i < R; i++) {
            if (board[i] == 1) {
                answer = 1
                break
            }
        }
    }
    for (let i = 0; i < R; i++) {
        if (board[0][i] == 1) {
            answer = 1
            break
        }
    }
    for (let r = 1; r < R; r++) {
        for (let c = 1; c < C; c++) {
            if (board[r][c] == 1)
                board[r][c] = 1 + Math.min(board[r - 1][c], board[r - 1][c - 1], board[r][c - 1])
            answer = Math.max(board[r][c], answer)
        }
    }
    return Math.pow(answer, 2)
}
